pyeudiw/storage/db_engine.py

```diff
@@ -83,10 +85,58 @@ def update_response_object(self, nonce: str, state: str, response_object: dict)
                     f"Cannot update document with nonce {nonce} and state {state} on {db_name}")
 
         if replica_count == 0:
-            raise Exception(
+            raise ReplicaError(
                 f"Cannot update document with state {state} and nonce {nonce} on any instance")
 
         return replica_count
+    
+    def find_chain(self, entity_id: str) -> Union[dict, None]:
+        for db_name, storage in self.storages:
+            try:
+                chain = storage.find_chain(entity_id)
+
+                if chain:
+                    return chain
+                
+            except Exception as e:
+                logger.critical(f"Error {str(e)}")
+                logger.critical(
+                    f"Cannot find chain {entity_id} on {db_name}")
+                
+        raise ReplicaError(f"Cannot find chain {entity_id} on any instance")
```

Comment:
this is not a Replica error since it doesn't take in account any replication mechanisms but just the missing trust attestation, please change accordingly this exception

Reply:
Resolved in [302d523](https://github.com/italia/eudi-wallet-it-python/pull/94/commits/302d523f5d36cafab1c3c04c049a27d06c21d575)